### PR TITLE
[ISSUE-450] Build custom kind for CI

### PR DESCRIPTION
**Pull request by safronovD**
### Issue #450 

Create new make targets:
- kind-build - patch and build custom kind version locally
- kind-create-cluster - create kind cluster using custom kind with kind.conf for CI
- kind-delete-cluster - delete kind cluster
- kind-install - update kind in /usr/bin (for internal use)

## PR checklist
- [x] Add link to the issue
- [x] Choose PR label
- [ ] New unit tests added
- [x] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
_Link to custom CI build_

### CODE REVIEW

File: test/kind/kind.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- ./kind-0.11.1/pkg/cluster/internal/providers/docker/provision.go	2021-05-28 02:18:13.000000000 +0300
++++ ./kind-0.11.1/pkg/cluster/internal/providers/docker/patched_provision.go	2021-06-30 17:25:58.317297835 +0300
+@@ -243,6 +243,8 @@
```
Comment (yurkov-anton) [inline test/kind/kind.patch:3]: So conceived?

**Review by yurkov-anton — APPROVED** [review]

File: test/kind/kind.patch (hunk 1 above)
Comment (safronovD) [reply test/kind/kind.patch:3]: yep :)

File: Makefile.validation [hunk 2]
```diff
@@ -109,3 +109,24 @@ kind-load-images:
...
+prepare-kind:
+	# Get kind sources
+	wget -O kind-src https://codeload.github.com/kubernetes-sigs/kind/tar.gz/refs/tags/v0.11.1
+	tar -xzvf kind-src
+
+	# Add "--ipc=host" docker option and build binary
+	patch -p0 < test/kind/kind.patch
+	cd kind-0.11.1 && make build
+
+	# Copy executive file
```
Comment (mishoyama) [inline Makefile.validation:122]: This should be a separate target. User might not have an access for this.

**Review by mishoyama — CHANGES_REQUESTED** [review]

File: Makefile.validation (hunk 2 above)
Comment (safronovD) [reply Makefile.validation:128]: created `kind-build` and `kind-copy`

**Comment by mishoyama** [comment]
Please add information to PR and issue description why we need this.

File: Makefile.validation (hunk 2 above)
Comment (mishoyama) [reply Makefile.validation:128]: It should be `kind-install`

File: Makefile.validation [hunk 3]
```diff
@@ -109,3 +109,26 @@ kind-load-images:
...
+
+prepare-kind: build-kind copy-kind
+
+build-kind:
+	# Get kind sources
+	wget -O kind-src https://codeload.github.com/kubernetes-sigs/kind/tar.gz/refs/tags/v0.11.1
+	tar -xzvf kind-src
+
+	# Add "--ipc=host" docker option and build binary
+	patch -p0 < test/kind/kind.patch
```
Comment (mishoyama) [inline Makefile.validation:121]: Can we patch version as well for user to see that kind is patched?

File: Makefile.validation [hunk 4]
```diff
@@ -109,3 +109,26 @@ kind-load-images:
...
+	cd kind-0.11.1 && make build
+
+	# Copy file
+	mv kind-0.11.1/bin/kind .
+
+copy-kind:
+	# Copy executive file
+	rm /usr/bin/kind
+	chmod +x ./kind
+	mv ./kind /usr/bin
```
Comment (mishoyama) [inline Makefile.validation:131]: Shouldn't we fail when kind is already installed?

**Review by mishoyama — CHANGES_REQUESTED** [review]

**Review by mishoyama — APPROVED** [review]

**Pull request merged**